gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/iapi/sql/dictionary/StatementTablePermission.java

```diff
@@ -134,7 +135,9 @@ public void check( LanguageConnectionContext lcc,
 	{
 		DataDictionary dd = lcc.getDataDictionary();
 	
-		if( ! hasPermissionOnTable( dd, authorizationId, forGrant))
+		if( ! hasPermissionOnTable( dd, authorizationId, forGrant) && !(getTableDescriptor(dd)
+				.getSchemaName().equalsIgnoreCase(Misc.SNAPPY_HIVE_METASTORE) && this.privType ==
```

Comment:
Can we use the same method that is defined in above in StatementColumnPermission.java. May be move that method to this class since StatementColumnPermission derives from this class.